src/compiler/checker.ts

```diff
@@ -4151,13 +4151,13 @@ namespace ts {
             if (parentType === unknownType) {
                 return unknownType;
             }
-            // If no type was specified or inferred for parent, or if the specified or inferred type is any,
-            // infer from the initializer of the binding element if one is present. Otherwise, go with the
-            // undefined or any type of the parent.
-            if (!parentType || isTypeAny(parentType)) {
-                if (declaration.initializer) {
-                    return checkDeclarationInitializer(declaration);
-                }
+            // If no type was specified or inferred for parent,
+            // infer from the initializer of the binding element if one is present.
+            // Otherwise, go with the undefined type of the parent.
+            if (!parentType) {
+                return declaration.initializer ? checkDeclarationInitializer(declaration) : parentType;

```

Comment:
If there's no `parentType`, and no `declaration.initializer`, then we return `parentType` when we know it does not exist - it'd probably be nicer for future references to this code if it explicitly returned `undefined` and had `undefined` as part of it's return type signature, if you ever feel like coming back to this.